packages-exp/auth-exp/src/core/user/additional_user_info.ts

```diff
@@ -0,0 +1,108 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { AdditionalUserInfo, UserProfile } from '../../model/user';
+import { IdTokenResponse, IdTokenResponseKind } from '../../model/id_token';
+import { ProviderId } from '../providers';
+import { parseToken } from './id_token_result';
+
+export function fromIdTokenResponse(
+  idTokenResponse: IdTokenResponse,
+): AdditionalUserInfo | null {
+  const { providerId } = idTokenResponse;
+  const profile =
+    typeof idTokenResponse.rawUserInfo === 'string'
```

Comment:
is rawUserInfo ever not a string? should we have an assertion here?

Reply:
Thats fine yeah. We actually just need to do a null check.